### ISSUE TITLE
How to toggle "topmost" of a window while program is running?

### ISSUE THREAD

**Issue by NadyaNayme** [gabdube/native-windows-gui#245]
I'd like to allow users to control whether or not the window is `topmost`. I don't see an intuitive path to handling this.

After a lot of digging I naïvely thought I could do something like:

```
    fn set_topmost(app_rc: &Rc<my_app>) {
        use winapi::um::winuser::SetWindowPos;
        use winapi::um::winuser::{HWND_TOPMOST, SWP_NOSIZE, SWP_NOMOVE, SWP_NOACTIVATE};
        unsafe { SetWindowPos(app_rc.window.handle.hwnd().unwrap(), HWND_TOPMOST, 0, 0, 0, 0, SWP_NOMOVE | SWP_NOSIZE | SWP_NOACTIVATE); };
        app_rc.window.invalidate();
    }
```

Which compiles but also does nothing because I'm not well versed enough to really understand what I'm doing. ^^; I was hoping it would update the window positioning with `HWND_TOP` or `HWND_TOPMOST` and then invalidating it would force a redraw which would account for the changed flag. That's not what happens because I'm probably doing something silly/stupid.

The other alternative I can think of is to spawn two windows - one visible and one not - with different `topmost` values and share state between them hiding one and showing the other but that seems like a total headache/mess waiting to happen.

**Comment by sollyucko** [comment]
> I'd like to allow users to control whether or not the window is `topmost`. I don't see an intuitive path to handling this.
> 
> After a lot of digging I naïvely thought I could do something like:
> 
> ```
>     fn set_topmost(app_rc: &Rc<my_app>) {
>         use winapi::um::winuser::SetWindowPos;
>         use winapi::um::winuser::{HWND_TOPMOST, SWP_NOSIZE, SWP_NOMOVE, SWP_NOACTIVATE};
>         unsafe { SetWindowPos(app_rc.window.handle.hwnd().unwrap(), HWND_TOPMOST, 0, 0, 0, 0, SWP_NOMOVE | SWP_NOSIZE | SWP_NOACTIVATE); };
>         app_rc.window.invalidate();
>     }
> ```
> 
> Which compiles but also does nothing because I'm not well versed enough to really understand what I'm doing. ^^; I was hoping it would update the window positioning with `HWND_TOP` or `HWND_TOPMOST` and then invalidating it would force a redraw which would account for the changed flag. That's not what happens because I'm probably doing something silly/stupid.
> 
> The other alternative I can think of is to spawn two windows - one visible and one not - with different `topmost` values and share state between them hiding one and showing the other but that seems like a total headache/mess waiting to happen.

As far as I understand, the `SetWindowPos` should take effect immediately. I'm not sure what `invalidate` in this crate does. In fact, maybe it's overwriting it? Maybe try using [a system call (syscall) monitoring tool](https://stackoverflow.com/questions/3847745/systrace-for-windows) to make sure the `SetWindowPos`/`HWND_TOPMOST` (not sure what the underlying syscall is) is actually getting called, and nothing else is getting called that's overriding it?

**Comment by NadyaNayme** [comment]
Apologies, I should have shared what I was _actually_ trying to do instead of the related thing to what I was trying to do.

If I had tried the inverse (starting from `topmost: false` and trying to set `topmost: true`) I would have realized sooner I wasn't using the correct flag to unset `topmost`. I was trying to set `HWND_TOP` instead of `HWND_NOTOPMOST` to get rid of `HWND_TOPMOST` which as you might expect doesn't work and I should learn to [read documentation more carefully](https://docs.microsoft.com/en-us/windows/win32/api/winuser/nf-winuser-setwindowpos).

The example I provided works. The example I _didn't_ provide was the issue here. What I should have done is shared the `unset_topmost` function which, as you might expect, tries to unset `topmost` on the Window.
